### PR TITLE
Various cleanup

### PR DESCRIPTION
**Pull request by ethansf**
This is kind of three pull requests in one, but I had to make these changes to get everything ready for production.

1. redis_cleanup - close redis socket connections better upon websocket disconnect
2. nginx_safe_response - when there was an exception before the websocket handshake, and we were using uwsgi, the http headers were not written.  This caused trouble when proxying behing nginx
3. Various error message improvements - added a little more info to the logger to help debugging

If you accept this, please ignore my other pull requests.
If this is too much, you can take one or two of the other pull requests without problem

### CODE REVIEW

File: ws4redis/websocket.py [hunk 1]
```diff
@@ -198,14 +198,14 @@ def receive(self):
             raise WebSocketError("Connection is already closed")
         try:
             return self.read_message()
-        except UnicodeError:
-            logger.info('websocket.receive: UnicodeError')
+        except UnicodeError as e:
+            logger.info(u'websocket.receive: UnicodeError {}'.format(e))
```
Comment (jrief) [inline ws4redis/websocket.py]: Please don't mark strings as unicode.
Instead add ``from __future__ import unicode_literals`` to the top of the file.
This makes a transition to Py3 much easier.

File: ws4redis/websocket.py [hunk 2]
```diff
@@ -198,14 +198,14 @@ def receive(self):
...
             self.close(1007)
-        except WebSocketError:
-            logger.info('websocket.receive: WebSocketError')
+        except WebSocketError as e:
+            logger.info(u'websocket.receive: WebSocketError {}'.format(e))
             self.close(1002)
         except Exception as e:
-            logger.info('websocket.receive: Unknown error %s', e)
+            logger.info(u'websocket.receive: Unknown error {}'.format(e))
```
Comment (jrief) [inline ws4redis/websocket.py]: Just a note. I pilfered this file, therefore the old string formatting was used. Is this file now Py3 safe?

Comment (ethansf) [inline ws4redis/websocket.py]: I made the fixes in commit 969e702.  Maybe this is now safe for merging?
If you don't like them, you can also ignore these changes, but they were
nice in local debugging.

The most important ones in my eyes where the redis_cleanup, which was
already merged, and the nginx_safe_response.  As we had some interesting
hanging connections in a production environment when I raised
PermissionDenied.

By the way, this is a pretty awesome package.  In one day I got the
websocket server written and unit tested.  Production loadtesting and
playing well with the clients took a bit longer though.  I replaced a
unmaintable homebrew tornado app with this.

Cheers,
 Ethan



2015-04-23 17:52 GMT+02:00 Jacob Rief <notifications@github.com>:

Comment (ethansf) [inline ws4redis/websocket.py]: Also, I noticed that all the CI builds have failed for Django1.7 (which
they only started testing with build #130).  Seems like the fixtures don't
work.  Are you working on fixing this or "get around to it sometime"? (Cuz
I we are porting our app to 1.7 in the next month...)

Cheers,
 Ethan

2015-04-23 18:20 GMT+02:00 Ethan Snyder-Frey <ethan@orderbird.com>:

**Comment by jrief** [comment]
It would be great if you can have a look at the fixture causing the CI test to fail.
I'm currently out of resources to fix this.

**Pull request merged**